calicoctl/calico_ctl/diags.py

```diff
@@ -50,19 +45,24 @@ def diags(arguments):
     this file's docstring with docopt
     :return: None
     """
-    print("Collecting diags")
+    print("Collecting diagnostics")
+    save_diags(arguments["--log-dir"])
+    sys.exit(0)
 
-    # Check runtime.  
-    runtime = arguments.get("--runtime")
-    if not runtime in ["docker", "rkt"]:
-        print "Invalid runtime specified: '%s'" % runtime
-        sys.exit(1)
 
-    save_diags(arguments["--log-dir"], runtime)
-    sys.exit(0)
+def write_diags(f, command):
```

Comment:
I was thinking that DiagsFileWriter could have additional methods for writing and writing and running command.  That way the file object is completely wrapped up in the file writer.

Not too fussed either way tbh.